Source/swiftlint/Helpers/LintableFilesVisitor.swift

```diff
@@ -162,25 +163,64 @@ struct LintableFilesVisitor {
         return nil
     }
 
-    private static func loadCompileCommands(_ path: String) -> [String: [String]]? {
-        guard let jsonContents = FileManager.default.contents(atPath: path),
-            let object = try? JSONSerialization.jsonObject(with: jsonContents),
+    private static func loadCompileCommands(_ path: String) -> Result<[File: Arguments], CompileCommandsLoadError> {
+        guard let jsonContents = FileManager.default.contents(atPath: path) else {
+            return .failure(.nonExistantFile(path))
+        }
+
+        guard let object = try? JSONSerialization.jsonObject(with: jsonContents),
             let compileDB = object as? [[String: Any]] else {
-            return nil
+            return .failure(.malformedCommands(path))
         }
 
         // Convert the compilation database to a dictionary, with source files as keys and compiler arguments as values.
         //
         // Compilation databases are an array of dictionaries. Each dict has "file" and "arguments" keys.
-        return compileDB.reduce(into: [:]) { (commands: inout [File: Arguments], entry: [String: Any]) in
```

Comment:
I wasn't happy swapping this reduce to a for loop, but chose to do this over throwing since `Result` is more widely used in this codebase AFAICT

Reply:
I'm happy with the change.